### ISSUE TITLE
Question about the performance

### ISSUE THREAD

**Issue by Charlie1994** [yongqyu/STRNN#3]
Hi Yongqyu,

Thank you for re-implementing ST-RNN.  I am quite interested in it. However, I find that the recall performances are largely lower than the figures given in the paper.  

My results (15 epochs):
('recall@1: ', 0.0010507880910683013)
('recall@5: ', 0.0051138353765324)
('recall@10: ', 0.00903677758318739)
('recall@100: ', 0.025218914185639228)
('recall@1000: ', 0.0533800350262697)
('recall@10000: ', 0.09936952714535902)

Results in the paper:
recall@1 0.0304,
recall@5 0.1524
recall@10 0.2714

It seems that the model is overfitting. And I noticed that your regulation parameter is set to be 0.0005, but the author set this parameter to be 0.01. I'll try 0.01 shortly.

By the way, what is your opinion about this performance difference?

Thanks,
Charles

**Comment by yongqyu** [comment]
Hi. Charles.
First of all, thank you for your interest in my implementation.
This implementation code is old and I can not remember exactly, but the
most problematic part of the implementation of this paper was the linear
interpolation for the transition matrices.
Since the upper bound and low bound are not specified in the paper, I set
them arbitrarily.
And thank you for your weight decay advice. From this advice, other
hyper-parameters may also be a source of performance differences.
I'll also try it out.

If you have any other questions, please contact me again.
Thank you.

2018년 10월 18일 (목) 오전 9:03, Charles Li <notifications@github.com>님이 작성:
